pkg/cloud-controller-manager/ccm.go

```diff
@@ -58,49 +61,45 @@ func newCloudProvider(reader io.Reader) (cloudprovider.Interface, error) {
 
 	namespace := rawConfig.Contexts[rawConfig.CurrentContext].Namespace
 
-	loadBalancerManager, err := newLoadBalancerManager(clientConfig, namespace)
+	localCfg, err := kubeconfig.GetNonInteractiveClientConfig(os.Getenv("KUBECONFIG")).ClientConfig()
 	if err != nil {
-		return nil, fmt.Errorf("create load balancer manager faield, err: %w", err)
+		return nil, err
 	}
 
-	instanceManager, err := newInstanceManager(clientConfig, namespace)
-	if err != nil {
-		return nil, fmt.Errorf("create instance manager failed, error: %w", err)
-	}
+	cp := &CloudProvider{
+		localCoreFactory: ctlcore.NewFactoryFromConfigOrDie(localCfg),
+		lbFactory:        ctllb.NewFactoryFromConfigOrDie(clientConfig),
+		kubevirtFactory:  ctlkubevirt.NewFactoryFromConfigOrDie(clientConfig),
```

Comment:
The cloud-provider-harvester log displays some errors:
```
E0508 00:27:48.659964    7682 reflector.go:140] github.com/rancher/lasso/pkg/cache/cache.go:145: Failed to watch *v1.VirtualMachineInstance: failed to list *v1.VirtualMachineInstance: virtualmachineinstances.kubevirt.io is forbidden: User "system:serviceaccount:default:abcd" cannot list resource "virtualmachineinstances" in API group "kubevirt.io" at the cluster scope
W0508 00:27:54.670001    7682 reflector.go:424] github.com/rancher/lasso/pkg/cache/cache.go:145: failed to list *v1.VirtualMachineInstance: virtualmachineinstances.kubevirt.io is forbidden: User "system:serviceaccount:default:abcd" cannot list resource "virtualmachineinstances" in API group "kubevirt.io" at the cluster scope
E0508 00:27:54.670030    7682 reflector.go:140] github.com/rancher/lasso/pkg/cache/cache.go:145: Failed to watch *v1.VirtualMachineInstance: failed to list *v1.VirtualMachineInstance: virtualmachineinstances.kubevirt.io is forbidden: User "system:serviceaccount:default:abcd" cannot list resource "virtualmachineinstances" in API group "kubevirt.io" at the cluster scope
W0508 00:28:07.171436    7682 reflector.go:424] github.com/rancher/lasso/pkg/cache/cache.go:145: failed to list *v1.VirtualMachineInstance: virtualmachineinstances.kubevirt.io is forbidden: User "system:serviceaccount:default:abcd" cannot list resource "virtualmachineinstances" in API group "kubevirt.io" at the cluster scope
E0508 00:28:07.171484    7682 reflector.go:140] github.com/rancher/lasso/pkg/cache/cache.go:145: Failed to watch *v1.VirtualMachineInstance: failed to list *v1.VirtualMachineInstance: virtualmachineinstances.kubevirt.io is forbidden: User "system:serviceaccount:default:abcd" cannot list resource "virtualmachineinstances" in API group "kubevirt.io" at the cluster scope
W0508 00:28:24.326629    7682 reflector.go:424] github.com/rancher/lasso/pkg/cache/cache.go:145: failed to list *v1.VirtualMachineInstance: virtualmachineinstances.kubevirt.io is forbidden: User "system:serviceaccount:default:abcd" cannot list resource "virtualmachineinstances" in API group "kubevirt.io" at the cluster scope
E0508 00:28:24.326656    7682 reflector.go:140] github.com/rancher/lasso/pkg/cache/cache.go:145: Failed to watch *v1.VirtualMachineInstance: failed to list *v1.VirtualMachineInstance: virtualmachineinstances.kubevirt.io is forbidden: User "system:serviceaccount:default:abcd" cannot list resource "virtualmachineinstances" in API group "kubevirt.io" at the cluster scope
...
```

Since the downstream cluster can only access resources within a specific namespace in Harvester, the `kubevirtFactory` definition should pass the namespace to the `Factory`, eg:
```
kubevirtFactory := ctlkubevirt.NewFactoryFromConfigWithOptionsOrDie(clientConfig, &ctlkubevirt.FactoryOptions{
	Namespace: namespace,
})
```